code/game/machinery/autolathe.dm

```diff
@@ -184,14 +239,20 @@
 		if(istype(O, /obj/item/disk/design_disk))
 			var/obj/item/disk/design_disk/D = O
 			if(D.blueprint)
+				if(!(D.blueprint.build_type & AUTOLATHE)) // otherwise, would silently fail in AddDesign2Known
+					to_chat(user, "<span class='warning'>This design is not compatible with the autolathe.</span>")
+					return 1
 				user.visible_message("[user] begins to load \the [O] in \the [src]...",
 					"You begin to load a design from \the [O]...",
 					"You hear the chatter of a floppy drive.")
 				playsound(get_turf(src), 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
-				busy = 1
+				busy = TRUE
 				if(do_after(user, 14.4, target = src))
+					if(!("Imported" in D.blueprint.category)) // R&D should always ensure this is set on design disks, but it doesn't.
+						D.blueprint.category += "Imported" // now it will actually show up in the list.
 					files.AddDesign2Known(D.blueprint)
-				busy = 0
+					SStgui.close_uis(src) // forces all connected users to re-open the TGUI. Imported entries won't show otherwise due to static_data
```

Comment:
Try using `SStgui.update_uis(src)` here as that *should* update static data.

Reply:
update_uis(src) does not reload static data. Only closing the entire window and re-opening it does.